src/applications/vaos/new-appointment/redux/actions.js

```diff
@@ -258,53 +264,165 @@ export function fetchFacilityDetails(facilityId) {
   };
 }
 
+export function checkEligibility(location, siteId) {
+  return async (dispatch, getState) => {
+    const state = getState();
+    const useVSP = vaosVSPAppointmentNew(state);
+    const directSchedulingEnabled = vaosDirectScheduling(state);
+    const typeOfCareId = getTypeOfCare(getState().newAppointment.data)?.id;
+
+    dispatch({
+      type: FORM_ELIGIBILITY_CHECKS,
```

Comment:
I like having this as a separate function, but I don't think it should have its own action, since this logic happens directly from either the form data update action or the form page open action for the new facility page.

Reply:
Since this has been moved to the continue button, I think it's okay for this it's own action, right? It'll be dispatched from the `next()` of typeOfFacilityV2 in the newAppointmentFlow file.

Reply:
Technically in that case you'd want it within the page change progress actions. The action the user took is to change the page/continue, rather than check eligibility.

I do see that it's easier to do it this way now that it's in the `next()` function, so I'd probably just leave it for now and we can revisit it later if we want.